blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java

```diff
@@ -72,6 +89,16 @@ public String getDisplayDescription() {
     }
 
     @Override
+    public BlueTestResultContainer getTests() {
+        return new BlueTestResultContainerImpl(this, run);
+    }
+
+    @Override
+    public BlueTestSummary getTestSummary() {
```

Comment:
This can result in to performance issues. test results are read off disk. So far flow graph returned thru nodes API is pretty light weight, embedding test summary could be expensive for large number of nodes. 

I would like @svanoort to weigh in here too, specially keeping in mind perf issues discussed at: https://github.com/jenkinsci/blueocean-plugin/pull/1517#issuecomment-341268422.

Reply:
I'm going to defer to you and @svanoort on performance matters.

Reply:
If I'm not mistaken, we're only reading from disk once - once the action's in memory, we shouldn't be reading from disk again, should we?